### DISCUSSION TITLE
Search Integration

### DISCUSSION THREAD

**Discussion by trailjeep** [TriliumNext/Trilium#4481] [Ideas]
I run Trilium server as a docker instance for my notes.
I settled on [linkding](https://github.com/sissbruecker/linkding) for self-hosted bookmarks and I found this: [linkding injector](https://addons.mozilla.org/en-US/firefox/addon/linkding-injector/) (Injects search results from the linkding bookmark service into search pages like google and duckduckgo.)
![awesomeness](https://addons.mozilla.org/user-media/previews/full/265/265312.png)
For me it's a game-changer to search multiple personal data sources at once!  It's especially useful when trying to recall something I want but can't quite put my finger on it. Yes, getting old is fun!
This would be an incredibly useful browser extension for Trilium (I would be happy to contribute to a bounty.)
I think it should be doable for the server version, but not sure about desktop.

**Comment by meichthys** [comment]
This is a neat idea that I have not thought of. It seems like this should be able to be implemented via the ETAPI. 
It would be interesting to look into adding a Trilium instance as a search engine to a meta search engine like searxng.

**Comment by zerebos** [comment]
Definitely possible to do something similar where the extension gets pointed to a personal instance. It would definitely be something fun to work on if I had the time.

**Comment by Nriver** [comment]
`Evernote Web Clipper` has this feature for Evernote. https://help.evernote.com/hc/en-us/articles/208313708-Evernote-Web-Clipper-related-results
